### PR TITLE
mgmt, support getNetworkInterfaceAsync

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

fIx https://github.com/Azure/azure-sdk-for-java/issues/25398

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java [hunk 1]
```diff
@@ -609,6 +609,15 @@ public VirtualMachineScaleSetNetworkInterface getNetworkInterfaceByInstanceId(St
             .getByVirtualMachineScaleSetInstanceId(this.resourceGroupName(), this.name(), instanceId, name);
     }
 
+    @Override
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java:612]: Have a test to cover this interface?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java:612]: Tested locally, but not added to test, as basically parallel code.

**Review by haolingdong-msft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java:612]: Added the one line test on the async.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 2]
```diff
@@ -291,13 +286,14 @@ public void canCreateVirtualMachineScaleSetWithCustomScriptExtension() throws Ex
...
-            Assertions.assertEquals(response.getStatusCode(), 200);
-        }
+        Assertions.assertNotNull(fqdn);
+//        // Assert public load balancing connection
+//        if (!isPlaybackMode()) {
+//            HttpClient client = HttpClient.createDefault();
+//            HttpRequest request = new HttpRequest(HttpMethod.GET, "http://" + fqdn);
+//            HttpResponse response = client.send(request).block();
+//            Assertions.assertEquals(response.getStatusCode(), 200);
+//        }
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:296]: Remove due to these no longer work (could be caused by security policy).

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 3]
```diff
@@ -1373,7 +1369,9 @@ private void checkVMInstances(VirtualMachineScaleSet vmScaleSet) {
                 vmScaleSet.getNetworkInterfaceByInstanceId(vm.instanceId(), nic.name());
             Assertions.assertNotNull(nicA);
             VirtualMachineScaleSetNetworkInterface nicB = vm.getNetworkInterface(nic.name());
+            String nicIdB = vm.getNetworkInterfaceAsync(nic.name()).map(n -> nic.primaryIPConfiguration().networkId()).block();
             Assertions.assertNotNull(nicB);
+            Assertions.assertNotNull(nicIdB);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:1374]: Added one line test.

**Pull request merged**